proposals/lip-0050.md

```diff
@@ -171,14 +171,36 @@ registerBLSKeyParamsSchema = {
         "proofOfPossession": {
             "dataType": "bytes",
             "fieldNumber": 2
+        },
+        "generatorKey": {
+            "dataType": "bytes",
+            "fieldNumber": 3
         }
     }
 }
 ```
 
 ##### Execution
 
-Executing a transaction `trs` triggering an register BLS key command is done by calling the function `setValidatorBLSKey(validatorAddress, trs.params.proofOfPossession, trs.params.blsKey)` where `validatorAddress` is the 20-byte address derived from `trs.senderPublicKey` and `setValidatorBLSKey` is defined in the [Validators module][lip-0044#setValidatoBLSKey]. If this function returns false, the transaction is invalid.
+Executing a transaction `trs` triggering an register keys command is done by calling the logic below
+
+```python
+validatorAddress = 20-byte address derived from trs.senderPublicKey
```

Comment:
already going in the direction of having valid pytohn code, we could have something like `trs.senderPublicKey.to_address()`, where the sender public key is an instance of the public key class, which has this functionality (see, eg.: https://github.com/ricott1/dingus/blob/main/dingus/types/keys.py#L22)

Reply:
yes, that's a nice idea. Though I think I would do this change when we update to the more valid python code.

Reply:
After yesterday's meeting, i would leave it as is.